From a683ced1a2750da6794f4401ace89e5391523221 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 27 Oct 2024 01:49:31 +0200
Subject: [PATCH] Include icons as GResource

Add application-specific icons into the GResource, which is the usual place
for them.
---
 data/icons/meson.build                    | 21 ---------------------
 data/info.febvre.Komikku.gresource.xml.in |  6 ++++++
 2 files changed, 6 insertions(+), 21 deletions(-)

diff --git a/data/icons/meson.build b/data/icons/meson.build
index ce788acd..211277e6 100644
--- a/data/icons/meson.build
+++ b/data/icons/meson.build
@@ -23,24 +23,3 @@ install_data (
     install_dir: symbolic_dir,
     rename: '@0@-symbolic.svg'.format(app_id)
 )
-
-install_data (
-    'brush-symbolic.svg',
-    install_dir: symbolic_dir
-)
-install_data (
-    'computer-fail-symbolic.svg',
-    install_dir: symbolic_dir
-)
-install_data (
-    'explore-symbolic.svg',
-    install_dir: symbolic_dir
-)
-install_data (
-    'filter-symbolic.svg',
-    install_dir: symbolic_dir
-)
-install_data (
-    'library-symbolic.svg',
-    install_dir: symbolic_dir
-)
diff --git a/data/info.febvre.Komikku.gresource.xml.in b/data/info.febvre.Komikku.gresource.xml.in
index 401ec3b3..20a84c35 100644
--- a/data/info.febvre.Komikku.gresource.xml.in
+++ b/data/info.febvre.Komikku.gresource.xml.in
@@ -3,6 +3,12 @@
     <gresource prefix="/info/febvre/Komikku">
         <file compressed="true" preprocess="xml-stripblanks" alias="appdata.xml">@app_id@.appdata.xml</file>
 
+        <file alias="icons/scalable/apps/brush-symbolic.svg" preprocess="xml-stripblanks">icons/brush-symbolic.svg</file>
+        <file alias="icons/scalable/apps/computer-fail-symbolic.svg" preprocess="xml-stripblanks">icons/computer-fail-symbolic.svg</file>
+        <file alias="icons/scalable/apps/explore-symbolic.svg" preprocess="xml-stripblanks">icons/explore-symbolic.svg</file>
+        <file alias="icons/scalable/apps/filter-symbolic.svg" preprocess="xml-stripblanks">icons/filter-symbolic.svg</file>
+        <file alias="icons/scalable/apps/library-symbolic.svg" preprocess="xml-stripblanks">icons/library-symbolic.svg</file>
+
         <file compressed="true" preprocess="xml-stripblanks">ui/application_window.ui</file>
         <file compressed="true" preprocess="xml-stripblanks">ui/card.ui</file>
         <file compressed="true" preprocess="xml-stripblanks">ui/card_chapters_list_row.ui</file>
-- 
2.45.2

